server/metadata/key_encoder.go

```diff
@@ -16,13 +16,21 @@ package metadata
 
 import (
 	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/tigrisdata/tigris/errors"
 	"github.com/tigrisdata/tigris/internal"
 	"github.com/tigrisdata/tigris/keys"
 	"github.com/tigrisdata/tigris/schema"
 )
 
+type CacheEncoder interface {
+	EncodeCacheTableName(ns uint32, db uint32, name string) (string, error)
```

Comment:
Should we call `project` instead of `db` here?

Reply:
sure